### PR TITLE
US13-Task45 Integrate UI with Project api

### PR DESCRIPTION
**Pull request by shahih25**
US13-Task45 Integrate UI with Project api

### CODE REVIEW

File: taigaProject/src/taigaApi/controller/project_controller.py [hunk 1]
```diff
@@ -7,12 +8,12 @@
 
 router = APIRouter()
 
-@router.get("/Project")
-def auth(projectRequest: ProjectRequest):
-    project_info = get_project_by_slug(projectRequest.projectslug, projectRequest.authtoken)
+@router.post("/project")
```
Comment (achaur11) [inline taigaProject/src/taigaApi/controller/project_controller.py]: Keep it as Project, APIs should be in caps

File: taigaProject/src/taigaApi/controller/project_controller.py [hunk 2]
```diff
@@ -7,12 +8,12 @@
...
+def auth(projectRequest: ProjectRequest, token: Annotated[str | None, Header()] = None):
```
Comment (achaur11) [inline taigaProject/src/taigaApi/controller/project_controller.py:12]: What is the need of this change? Is it for a front end change?

Comment (shahih25) [reply taigaProject/src/taigaApi/controller/project_controller.py:12]: Token are generally sent as part of header. That's why I made this change and removed the field from projectRequest.

File: taigaProject/src/taigaApi/controller/project_controller.py (hunk 1 above)
Comment (shahih25) [reply taigaProject/src/taigaApi/controller/project_controller.py]: Updated. However, I found most of the website suggesting to use lowercase in the urls. Even Google search does that unless it's a query param.

File: taigaProject/src/taigaApi/controller/metric_controller.py [hunk 3]
```diff
@@ -7,11 +8,11 @@
 
 router = APIRouter()
 
-@router.get("/metric/LeadTime")
-def auth(projectRequest: ProjectRequest):
+@router.post("/metric/LeadTime")
```
Comment (shahih25) [inline taigaProject/src/taigaApi/controller/metric_controller.py:11]: Reason for post as we are expecting some data from user in the form of body.

File: taigaProject/src/taigaApi/controller/metric_controller.py [hunk 4]
```diff
@@ -7,11 +8,11 @@
...
+def get_lead_time_metric(projectRequest: ProjectRequest, token: Annotated[str | None, Header()] = None):
```
Comment (shahih25) [inline taigaProject/src/taigaApi/controller/metric_controller.py:12]: Token is passed as header similar to what taigaApi does.

File: taigaProject/src/taigaApi/task/getTaskHistory.py [hunk 5]
```diff
@@ -68,7 +68,7 @@ def get_task_lead_time(project_id, auth_token):
             "taskId": task["id"],
             "startTime": task["created_date"],
             "endTime": task['finished_date'],
-            "timeTaken": lead_time
+            "timeTaken": (finished_date - created_date).days
```
Comment (shahih25) [inline taigaProject/src/taigaApi/task/getTaskHistory.py:71]: Earlier we were adding the difference to leadTime. We just need the difference between start and finish time for a particular task.

File: taigaProject/src/taigaApi/task/getTasks.py [hunk 6]
```diff
@@ -19,6 +19,7 @@ def get_tasks(project_id, auth_token):
     headers = {
         'Authorization': f'Bearer {auth_token}',
         'Content-Type': 'application/json',
+        "x-disable-pagination": "True"
```
Comment (shahih25) [inline taigaProject/src/taigaApi/task/getTasks.py:22]: Disabled the pagination to get all the tasks and stories.

**Comment by shahih25** [comment]
Screenshot 2024-02-10 at 7 25 00 PM

**Comment by shahih25** [comment]
Screenshot 2024-02-10 at 10 13 06 PM

**Comment by spark290** [comment]
US13-Task45 PR review
Works well

**Pull request merged**